### ISSUE TITLE
Curl Options broken as array merge does not preserve keys

### ISSUE THREAD

**Issue by ajbonner** [sendgrid/php-http-client#12]
#### Issue Summary

I noticed some code I had depending on the sendgrid-php library suddenly started emitting output after a composer update. Turns out there's this code in Client.php

``` php
        curl_setopt_array($curl, array_merge([
            CURLOPT_RETURNTRANSFER => true,
            CURLOPT_HEADER => 1,
            CURLOPT_CUSTOMREQUEST => strtoupper($method),
            CURLOPT_SSL_VERIFYPEER => false,
        ], $this->curlOptions));

```

array_merge does not maintain keys, therefore the curl constants do not get set on the merged options array and return transfer is therefore false i.e. curl emits its body text response as if it had been echo'd out.
#### Steps to Reproduce
1. Perform any http request using library that has body content returned to client.
#### Technical details:

Tested on PHP7 with release sha#aa09fb89e2be2624de9db4dde914d27cbdece95c

**Comment by fernandoval** [comment]
Thanks,

I was investigating the issue since 09:00 (GMT -2).

I had the same problem when the Composer did the upgrade sendgrid-php to 5.1.1 and php-http-client to 5.5.0.

Now I downgraded to sendgrid-php 5.0.9 and php-http-client to 5.1.0, temporarily.

**Comment by ajbonner** [comment]
The fix is pretty simple, just use the array union operator [ ... ] + $this->curlOptions, rather than array_merge.

**Comment by thinkingserious** [comment]
This issue has been classified as a bug and is on our backlog for a fix.

Thanks!

**Comment by jakubkratina** [comment]
@thinkingserious When do you plan to release the bugfix? Thank you.

**Comment by thinkingserious** [comment]
@jakubkratina,

Unfortunately, not any time soon, unless we get a PR. This item is fairly low in the backlog right now :(

**Comment by ajbonner** [comment]
The PHP library straight up doesn't work with this bug.

Also there is already a PR see - #13

**Comment by jakubkratina** [comment]
@thinkingserious as @ajbonner said, there already is a PR :) Until merge I cannot use this package in the CLI :/

**Comment by thinkingserious** [comment]
@ajbonner, @jakubkratina,

The CLA is now received and that issue should be resolved shortly. Thanks for taking the time to let us know how important this fix is!

**Comment by jakubkratina** [comment]
@thinkingserious Thank you for your time

**Comment by adduc** [comment]
Note that without this change, requests with method DELETE are sent as GET instead (due to array_merge rekeying curl options).

Please consider an expedited merge and release, as the dependent client sendgrid/sendgrid-php uses the current bugged release.

**Comment by thinkingserious** [comment]
Fixed with this release: https://github.com/sendgrid/php-http-client/releases/tag/v3.5.1

**Comment by ajbonner** [comment]
Thanks @thinkingserious appreciate the quick merge!

**Comment by upchuk** [comment]
@thinkingserious This issue has been reintroduced with 3.9.0...

**Comment by thinkingserious** [comment]
Thanks for the heads up @upchuk!

The fix will be implemented shortly in 3.9.2
